src/arraymancer/tensor/private/p_display.nim

```diff
@@ -14,8 +14,64 @@
 
 import  ../../private/functional, ../higher_order_applymap,
         ../shapeshifting, ../data_structure,
-        ../accessors,
-        sequtils, strutils
+        ../accessors
+import std / [sequtils, strutils, strformat, typetraits]
+
+# We use different specifier string "tokens" to set the tensor display mode:
+# - || or <>||: forces "table mode" (without or with header)
+# - [:] or <>[:] (or the equivalent <:>): multi line array mode (w/o or w/ header)
+# - [] or <>[] (or the equivalent <>): single line array mode (w/o or w/ header)
+# If none of those tokens is found use table mode
+#
+func isTensorHeaderEnabled(specifier: string): bool =
+  ## Return true when a header must be shown based on the format specifier
+  # Show a header when "<>" (i.e. the explicit header marker) or "<:>" (i.e. the
+  # combined <>[:] shortcut) is found
+  # Otherwise do not show a header if "||" or "[]" or "[:]" is found
+  # Show a header in all other cases
+  # if "<>" in specifier or "<:>" in specifier:
+  if "<" in specifier and ">" in specifier:
+    # Note that this also covers the "<>[]" and "<>||" cases
+    return true
+  elif specifier.count('|') == 2 or ("[" in specifier and "]" in specifier):
+    return false
+  return true
+
+type tensorDispMode = enum table, multi_line_array, single_line_array
+
+func getTensorDispMode(specifier: string): tensorDispMode =
+  ## Get the display mode based on the special "tensor specifier tokens"
+  let brackets = "[" in specifier and "]" in specifier
+  let colon_brackets = specifier.count(':') == 2
```

Comment:
Why do we test for 2 `:` here? Shouldn't it be one?

Reply:
"brackets" is when you use [] while "colon brackets" is when you use :: that's why we must check for a count of 2 colons.